### PR TITLE
Enumerate reports for each node

### LINKED ISSUE #2: List reports for each node
Under the node, we should have a `reports` directory that lists reports by their timestamp. The contents of the file should be the raw JSON output.

### PR DESCRIPTION
**Pull request by MikaelSmith**
Adds reports for each node. Reports are named based on the timestamp of
when the run ended. Select report properties are included as metadata
for searching, but the bulk of the report is only retrieved when reading
it because events may be a large field.

Resolves #2.

Signed-off-by: Michael Smith <michael.smith@puppet.com>

### CODE REVIEW

**Review by timidri — APPROVED** [review]
LGTM

File: puppetwash.rb [hunk 1]
```diff
@@ -152,6 +153,69 @@ def read
...
+  'environment': 'string',
+  'status': 'string',
+  'noop': 'boolean',
+  'puppet_version': 'string',
+  'producer': 'string',
+  'hash': 'string'
+}
+
+class ReportsDir < Wash::Entry
+  label 'reports_dir'
```
Comment (ekinanp) [inline puppetwash.rb:168]: @timidri One small nit I had was `facts_dir`; `reports_dir` -- I think the `_dir` is redundant. Are there any issues with changing them to `facts`; `reports` ? I don't think the Wash user will have a hard time figuring out that those are directories.

**Review by ekinanp — APPROVED** [review]

File: puppetwash.rb (hunk 1 above)
Comment (timidri) [reply puppetwash.rb:168]: Personally I find the explicit naming convention even easier to follow, but it’s a matter of preference I guess

Comment (ekinanp) [reply puppetwash.rb:168]: Ok, was just curious.

**Pull request merged**